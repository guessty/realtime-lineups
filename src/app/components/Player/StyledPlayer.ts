import styled from 'styled-components'

export default styled.div`
  text-align: center;

  .player {
    position: relative;
    padding: 2rem;
    font-size: 1rem;

    &__details {
      margin: 0 0 1rem;
      color: white;
      font-size: 1.6em;
      white-space: normal;
      overflow: hidden;
      text-overflow: ellipsis;
    }

    &__shirt {
      background: url('assets/player-shirt.png');
      background-repeat: no-repeat;
      background-size: 5em auto;
      position: relative;
      height: 5em;
      width: 5em;
      left: 50%;
      transform: translateX(-50%);
    }

    @media(max-width: 600px) {
      font-size: 0.9rem;
      padding: 2rem 1rem;
    }

    @media(max-width: 500px) {
      font-size: 0.8rem;
      padding: 2rem: 0.9rem;
    }
  }
`
