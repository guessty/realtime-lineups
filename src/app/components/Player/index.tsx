import * as React from 'react'
//
import StyledPlayer from './StyledPlayer'

interface PlayerProps {
  name: string
  position: string
  type:string
  formationPlace: string
}

const parseName = (name:string) => name.split(' ').slice(1).join(' ')

const Player = (props:PlayerProps) => (
  <React.Fragment>
    <StyledPlayer>
      <div className="player">
        <h4 className="player__details">{parseName(props.name)}<br/>({props.position})</h4>
        <div className="player__shirt" />
      </div>
    </StyledPlayer>
  </React.Fragment>
);

export default Player
