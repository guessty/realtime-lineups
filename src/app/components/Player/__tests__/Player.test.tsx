import * as React from 'react'
import * as Enzyme from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'
import * as renderer from 'react-test-renderer'
//
import Player from './../index'
// import { PITCH_FORMATIONS } from './../constants'

Enzyme.configure({adapter: new Adapter()})


describe('<Player />', () => {
  let component: any
  let componentWrapper: any
  const props = {
    name: 'player 1',
    position: 'GK',
    type: 'Goal Keeper',
    formationPlace: '1'
  }

  beforeEach(() => {
    component = (
      <Player {...props} />
    )
    componentWrapper = Enzyme.shallow(component, {})
  })

  it('should be defined', () => {
    expect(Player).toBeDefined()
  })

  it('should render', () => {
    expect(componentWrapper.length).toEqual(1)
  })

  it('capturing Snapshot of Player', () => {
    const renderedValue =  renderer.create(component).toJSON()
    expect(renderedValue).toMatchSnapshot();
  })
})

