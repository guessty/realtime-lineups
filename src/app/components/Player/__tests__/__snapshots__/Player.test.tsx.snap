// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Player /> capturing Snapshot of Player 1`] = `
<div
  className="sc-bdVaJa dnNKUV"
>
  <div
    className="player"
  >
    <h4
      className="player__details"
    >
      1
      <br />
      (
      GK
      )
    </h4>
    <div
      className="player__shirt"
    />
  </div>
</div>
`;
